src/server/node_manager.c

```diff
@@ -4555,15 +4353,33 @@ is_request(int stream, int version)
 			}
 		}
 
-		/* do not need to tdelete2() this stream, it was never inserted */
-		tpp_close(stream);
-
-		((mom_svrinfo_t *)(pmom->mi_data))->msr_stream = -1;
-		((mom_svrinfo_t *)(pmom->mi_data))->msr_state &= ~INUSE_INIT;
+		/* we save this stream for future communications */
+		psvrmom->msr_stream = stream;
+		psvrmom->msr_state &= ~INUSE_INIT;
+#ifdef NAS /* localmod 005 */
+		tinsert2((u_long)stream, 0ul, pmom, &streams);
+#else
+		tinsert2((u_long)stream, 0, pmom, &streams);
+#endif /* localmod 005 */
 
-		ping_a_mom(pmom, 1, 0);
+		tpp_eom(stream);
+		if (psvrmom->msr_vnode_pool > 0 ? psvrmom->reporting_mom : 1) {
```

Comment:
this is unnecessarily complex and doesn't really need 2 nested conditional statements, I think you can simply do this:
```suggestion
		if (psvrmom->msr_vnode_pool <= 0 || psvrmom->reporting_mom) {
```